torch/_inductor/kernel/mm.py

```diff
@@ -286,3 +287,25 @@ def tuned_mixed_mm(mat1, mat2, mat2_dtype):
             **mm_options(config, k, layout, b_prologue_cast_type),
         )
     return autotune_select_algorithm("mixed_mm", choices, [mat1, mat2], layout)
+
+
+def tuned_fused_int_mm_mul(mat1, mat2, mat3, out_dtype, *, layout=None):
+    out_dtype = (
+        torch.promote_types(mat3.get_dtype(), torch.int32)
```

Comment:
```suggestion
        functools.reduce(torch.promote_types, [
             mat1.get_dtype(),
             mat2.get_dtype(),
             mat3.get_dtype()
        ])
```
I am assuming this will match the semantics of `aten._int_mm`, but you should check to make sure.

Will aten._int_mm promote to torch.int32 even if none of the inputs are int32?  What about int64?

Reply:
int_mm only does int8 int8 -> int32 matmuls so i don't think those others are possible.

with torch.promote_types(mat3.get_dtype(), torch.int32) i was trying to set the output type to be what you'd naturally get from an int32 * float16 so that it matches the non torch-compiled model. What is the benefit of including the mat1 and mat2 dtypes?